pkgs/desktops/gnome/extensions/pano/default.nix

```diff
@@ -0,0 +1,138 @@
+{ lib
+, stdenv
+, atk
+, cogl
+, fetchFromGitHub
+, glib
+, gnome
+, gsound
+, gtk3
+, libgda
+, mkYarnModules
+, nodePackages
+, pango
+, substituteAll
+, yarn
+, yarn2nix
+}:
+
+let
+  version = "v6";
+  uuid = "pano@elhan.io";
+
+  girpathsPatch = (substituteAll {
+    src = ./deps/gnome-shell-extensions/pano/set-dirpaths.patch;
+    gda_path = "${libgda}/lib/girepository-1.0";
+    gsound_path = "${gsound}/lib/girepository-1.0";
+  });
+
+  panoSource = fetchFromGitHub {
+    owner = "oae";
+    repo = "gnome-shell-pano";
+    rev = "v${version}";
+    hash = "sha256-VFJcPemwlucwUOCW4CNit6EB2CAmX1kYQeziAYP03os=";
+  };
+
+  yarnNix = stdenv.mkDerivation {
+    pname = "gnome-shell-extension-pano-yarn-nix";
+    name = "gnome-shell-extension-pano-yarn-nix-${version}";
+    version = "${version}";
+    src = panoSource;
```

Comment:
This is working as expected when I build with `nix-build  -A gnomeExtensions.pano`.
However, `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"` produces:

    $ nix-env --option system x86_64-linux -f /home/michojel/.cache/nixpkgs-review/rev-f1da58b2d3f927fefd2d369692ebdc58fa9aee1f/nixpkgs -qaP --xml --out-path --show-trace --meta  
    error: path '/nix/store/l6c2m0ccyn2c45ryzpids07k4sjk9rdr-gnome-shell-extension-pano-yarn-nix-v6.drv' is not valid                                                              
                                                                                                                                                                               
           … while realising the context of path '/nix/store/cpq1jbx2p8s842aq4blwrnxx3jjs28p8-gnome-shell-extension-pano-yarn-nix-v6/yarn.nix'
           
Does anybody have a clue? Is there some better way how to build `yarn.nix` on the fly without manually copying `package.json` and `yarn.lock` files from the source?

Reply:
That is unfortunately possible in Nixpkgs yet. There was a [RFC for allowing import-from-derivation in Nixpkgs](https://github.com/NixOS/rfcs/pull/109) but that part got scraped, If I am reading that correctly.

Reply:
I see. Thank you for your help. I reverted back to using pre-generated yarn.nix.